lib/autoproj/ops/build.rb

```diff
@@ -84,7 +86,26 @@ def force_build_packages(selected_packages, all_enabled_packages)
             def build_packages(all_enabled_packages, options = Hash.new)
                 Autobuild.do_rebuild = false
                 Autobuild.do_forced_build = false
-                Autobuild.apply(all_enabled_packages, "autoproj-build", ['build'], options)
+                begin
+                    Autobuild.apply(all_enabled_packages, "autoproj-build", ['build'], options)
+                ensure
+                    build_report(all_enabled_packages)
+                end
+            end
+
+            TEMPLATES_DIR = File.join(File.dirname(__dir__), 'templates')
+
+            def build_report(package_list)
+                Autobuild.message "INFO: creating build report...."
+
+                log_dir = Autobuild.logdir
```

Comment:
From within autoproj, the log dir you should consider is `Workspace#log_dir`. The current workspace is accessible in the CLI classes as `#ws`, e.g. `ws.log_dir`